packages/blockchain-link-utils/src/solana.ts

```diff
@@ -0,0 +1,205 @@
+    const parsedInstructions = transaction.transaction.message.instructions.filter(
+        (instruction): instruction is ParsedInstruction => 'parsed' in instruction,
+    );
+
+    if (parsedInstructions.length === 0) {
+        return 'unknown';
+    }
+
+    const isTransfer = parsedInstructions.every(
+        instruction => instruction.parsed.type === 'transfer',
+    );
+
+    // for now we support only transfers, so we interpret all other transactions as `unknown`
+    if (isTransfer) {
+        if (
+            effects.length === 1 &&
+            effects[0]?.address === accountAddress &&
+            effects[0]?.amount.abs().isEqualTo(new BigNumber(transaction.meta?.fee || 0))
+        ) {
+            return 'self';
+        }
+
+        const senders = effects.filter(({ amount }) => amount.isNegative());
+
+        if (senders.find(({ address }) => address === accountAddress)) {
+            return 'sent';
+        }
+
+        const receivers = effects.filter(({ amount }) => amount.isPositive());
+
+        if (receivers.find(({ address }) => address === accountAddress)) {
+            return 'recv';
+        }
+    }
+
+    // TODO(vl): phase two, isTokenTx
+
+    return 'unknown';
+};
+
+export function getDetails(
+    transaction: ParsedTransactionWithMeta,
+    effects: TransactionEffect[],
+    accountAddress: string,
+): Transaction['details'] {
+    const senders = effects.filter(({ amount }) => amount.isNegative());
+    const receivers = effects.filter(({ amount }) => amount.isPositive());
+
+    const getVin = (effect: TransactionEffect, i: number) => ({
+        txid: transaction.transaction.signatures[0].toString(),
+        version: transaction.version,
+        isAddress: true,
+        isAccountOwned: effect.address === accountAddress,
+        n: i,
+        value: effect.amount.abs().toString(),
+        addresses: [effect.address],
+    });
+    return {
+        size: transaction.meta?.computeUnitsConsumed || 0,
+        totalInput: senders
+            .reduce((acc, curr) => acc.plus(curr.amount.abs()), new BigNumber(0))
+            .toString(),
+        totalOutput: receivers
+            .reduce((acc, curr) => acc.plus(curr.amount.abs()), new BigNumber(0))
+            .toString(),
+        vin: senders.map((sender, i) => getVin(sender, i)),
+        vout: receivers.map((receiver, i) => getVin(receiver, i)),
+    };
+}
+
+export function getAmount(
+    accountEffect: TransactionEffect | undefined,
+    txType: Transaction['type'],
+) {
+    if (!accountEffect) {
+        return '0';
+    }
+    if (txType === 'self') {
+        return accountEffect.amount?.abs().toString();
+    }
+    return accountEffect.amount.toString();
+}
+
+export const transformTransaction = (
+    tx: ParsedTransactionWithMeta,
+    accountAddress: string,
+    slotToBlockHeightMapping: Record<number, number | null>,
+): Transaction | null => {
```

Comment:
I am not particulary happy with the return type here. is this `| null` really needed? such return type does not match with return types of `transformTransaction` method for other coins.

Reply:
fixed in https://github.com/trezor/trezor-suite/pull/9498/commits/e6892fbed1330757fb6d9cb146cf79dae28d1f93